lib/camp.js

```diff
@@ -773,6 +774,38 @@ function pathLikeUnit(serverMethod) {
 var pathUnit = pathLikeUnit('path');
 var notFoundUnit = pathLikeUnit('notFound');
 
+// HTTP method unit (express-like, e.g. `camp.post('/', fn)`).
+function methodUnit (server) {
+  var paths = {};
+
+  function addMethod (method, path, handler) {
+    if (path[0] === '/') {
+      path = path.slice(1);
+    }
+    if (!(path in paths)) {
+      paths[path] = {};
+      server._method(path, function (req, res, next) {
+        var cb = paths[path][req.method.toLowerCase()];
+        if (cb) {
+          cb(req, res);
+        } else {
+          next();
+        }
+      });
+    }
+    paths[path][method] = handler;
+  }
+
+  var methods = ['get', 'post', 'patch', 'put', 'delete'];
```

Comment:
Indeed, express also implements a few more http://expressjs.com/en/guide/routing.html#route-methods. Maybe it makes sense to figure out the list of all methods we want to support.